malaria-ios/Managers/MedicineHelpers/MedicineManager.swift

```diff
@@ -0,0 +1,109 @@
+import Foundation
+
+class MedicineManager{
+    static let sharedInstance = MedicineManager()
+    
+    
+    
+    /// Setups a new pill and sets it as default
+    ///
+    /// If there is already a pill registed as default, it will no longer be the default pill
+    ///
+    /// :param: `Medicine.Pill`: the pill
+    /// :param: `NSDate`: fireDate
+    func setup(medicine : Medicine.Pill, fireDate: NSDate){
+        //unregister previous medicines
+        if let currentMed = getCurrentMedicine(){
+            currentMed.notificationManager.unsheduleNotification()
+        }
+        
+        Logger.Info("Storing new medicine")
+        registerNewMedicine(medicine)
+        setCurrentPill(medicine)
+        
+        Logger.Info("Setting up notification")
+        let newMed = getCurrentMedicine()!
+        
+        newMed.notificationManager.scheduleNotification(fireDate)
+
+        UserSettingsManager.setBool(UserSetting.DidConfiguredMedicine, true)
+        UserSettingsManager.syncronize()
```

Comment:
"The synchronize method, which is automatically invoked at periodic intervals, keeps the in-memory cache in sync with a user’s defaults database.".

At this case, is it safe? Should I call syncronize only when the application exists?